### PR TITLE
Stylefixes

### PR DESCRIPTION
**Pull request by dokato**
This addresses #212 - makes code coherent with new STYLEGUIDE.

**DoD**

- [x] Major project work has a corresponding task. If there’s no task for what you are doing, create it. Each task needs to be well defined and described.

- [x] Change has been tested (manually or with automated tests), everything runs correctly and works as expected. No existing functionality is broken.

- [x] No new error or warning messages are introduced.

- [x] All interaction with semantic functions, examples and docs are written from the perspective of the person using or receiving it. They are understandable and helpful to this person.

- [x] If the change affects code or repo structure, README, documentation and code comments should be updated. 

- [x] All code has been peer-reviewed before merging into any main branch.

- [x] All changes have been merged into the main branch we use for development (develop).

- [x] Continuous integration checks (linter, unit tests) are configured and passed.

- [x] Unit tests added for all new or changed logic.

- [x] All task requirements satisfied. The reviewer is responsible to verify each aspect of the task.

- [x] Any added or touched code follows our style-guide.

### CODE REVIEW

File: R/date_input.R [hunk 1]
```diff
@@ -43,9 +43,10 @@
 #'}
 #'
 #' @export
-date_input <- function(name, label = NULL, value = NULL, min = NULL, max = NULL,
+#' @rdname date_input
+date_input <- function(input_id, label = NULL, value = NULL, min = NULL, max = NULL,
```
Comment (olgamie) [inline R/date_input.R:47]: Is it possible to change the function, so it takes a character as an icon e.g. "dog" and do uiicon("dog") inside the function? We could then call shiny.semantic:::uiicon(), and don't export uiicon. Then the user would only have icon function exported.

File: R/dsl.R [hunk 2]
```diff
@@ -361,10 +356,10 @@ uimessage <- function(header, content, class = "", icon, closable = FALSE) {
 #'   shinyUI(
 #'     semanticPage(
 #'       title = "My page",
-#'       uimenu(menu_item("Menu"),
+#'       menu(menu_item("Menu"),
 #'              uidropdown(
```
Comment (olgamie) [inline R/dsl.R:360]: please fix examples, old uidropdown and uiicon

File: R/dsl.R [hunk 3]
```diff
@@ -430,10 +425,10 @@ menu_item <- function(..., item_feature = "", style = NULL, href = NULL) {
 #'
 #' @examples
 #'
-#' uidropdown(
+#' dropdown_menu(
```
Comment (olgamie) [inline R/dsl.R:428]: Please fix examples

File: R/input.R [hunk 4]
```diff
@@ -25,13 +25,15 @@ uiinput <- function(..., class = "") {
...
+#' @param input_id Input name. Reactive value is available under \code{input[[input_id]]}.
 #' @param value Pass value if you want to have default text.
 #' @param type Change depending what type of input is wanted. See details for options.
 #' @param placeholder Text visible in the input when nothing is inputted.
 #' @param attribs A named list of attributes to assign to the input.
+#' @param label character with label put above the input
+#' @param width The width of the input, eg. "40px"
 #'
 #' @details
 #' The following \code{type}s are allowed:
```
Comment (olgamie) [inline R/input.R:39]: spaces are missing between item{}(space missing here){}

File: R/input.R [hunk 5]
```diff
@@ -25,13 +25,15 @@ uiinput <- function(..., class = "") {
...
-#' under \code{input[[name]]}.
+#' under \code{input[[input_id]]}.
 #'
-#' @param name Input name. Reactive value is available under \code{input[[name]]}.
+#' @param input_id Input name. Reactive value is available under \code{input[[input_id]]}.
 #' @param value Pass value if you want to have default text.
 #' @param type Change depending what type of input is wanted. See details for options.
 #' @param placeholder Text visible in the input when nothing is inputted.
 #' @param attribs A named list of attributes to assign to the input.
+#' @param label character with label put above the input
```
Comment (olgamie) [inline R/input.R:35]: The label is put on the left side.

File: R/input.R [hunk 6]
```diff
@@ -52,33 +54,87 @@ uiinput <- function(..., class = "") {
 #' library(shiny.semantic)
 #'
 #' # Text input
-#' uiinput(
-#'   tags$label("Text input"),
-#'   uitextinput("ex", type = "text", placeholder = "Enter Text")
-#' )
+#' text_input("ex", label = "Your text", type = "text", placeholder = "Enter Text")
```
Comment (olgamie) [inline R/input.R:57]: Please change to
```
library(shiny)
library(shiny.semenitc)
if (interactive()) {
   ui <- semanticPage(
      uiinput(
         text_input("ex", label = "Your text", type = "text", placeholder = "Enter Text")
      )
   )
   server <- function(input, output, session) {
   }
   shinyApp(ui, server)
}
```

File: R/input.R [hunk 7]
```diff
@@ -99,15 +155,15 @@ uitextinput <- function(name, value = "", type = "text", placeholder = NULL, att
 #' # Text input
 #' uiinput(
 #'   tags$label("Numeric Input"),
-#'   uinumericinput("ex", 10)
+#'   numeric_input("ex", 10)
 #' )
 #'
 #' @export
-uinumericinput <- function(name, value, min = NA, max = NA, step = NA,
+numeric_input <- function(input_id, value, min = NA, max = NA, step = NA,
```
Comment (olgamie) [inline R/input.R:162]: `text_input` has a label argument. Should we make this consistent and add lebel here as well, or remove from text?

File: R/rating.R [hunk 8]
```diff
@@ -48,7 +62,7 @@ rating <- function(name, label = "", value = 0, max = 3, icon = "star",
 
 #' Update rating
 #'
-#' check \code{rating} to learn more.
+#' check \code{rating_input} to learn more.
```
Comment (olgamie) [inline R/rating.R:65]: This isn't very informative :D sth like: `Change the value of a rating input on the client` ?

File: R/slider.R [hunk 9]
```diff
@@ -62,24 +62,46 @@
...
+  form(
+    style = if (!is.null(width)) glue::glue("width: {shiny::validateCssUnit(width)};"),
+    label(label),
+    range_input(inputId, value, min, max, step = step)
+  )
+}
+
 #' Update slider Semantic UI component
 #'
 #' Change the value of a \code{\link{uislider}} input on the client.
```
Comment (olgamie) [inline R/slider.R:107]: Uislider => slider_input

File: R/slider.R [hunk 10]
```diff
@@ -88,20 +110,33 @@ uirange <- function(name, value, value2, min, max, step = 1, class = NULL) {
 #' @param name The id of the input object
 #' @param value The value to be selected for the sldier (lower value if using range).
 #'
-#' @seealso uislider
+#' @seealso slider
```
Comment (olgamie) [inline R/slider.R:113]: slider_input?

File: R/utils.R [hunk 11]
```diff
@@ -23,3 +23,24 @@ check_proper_color <- function(color) {
     invisible(color)
   }
 }
+
+#' Generates random id of element
```
Comment (olgamie) [inline R/utils.R:27]: Can we provide more details when to use it and example?

File: R/utils.R [hunk 12]
```diff
@@ -23,3 +23,24 @@ check_proper_color <- function(color) {
...
+#' @param id_length numeric with length of id (default 20)
+generate_random_id <- function(prefix, id_length = 20) {
+  random_id <- paste(sample(letters, id_length, replace = TRUE), collapse = "")
+  paste0(prefix, "-", random_id)
+}
+
+
+#' Check for extra arguments
+#'
+#' This throws warning if there're parameters not supported by semantic.
```
Comment (olgamie) [inline R/utils.R:39]: Same here can we add example

File: R/checkbox.R [hunk 13]
```diff
@@ -36,20 +36,20 @@
...
     tags$label(label)
   )
 }
 
-#' @rdname simple_checkbox
+#' @rdname checkbox
 #' @export
 toggle <- function(id, label = "", is_marked = TRUE, style = NULL) {
-  simple_checkbox(id, label, type = "toggle", is_marked = is_marked, style = style)
+  checkbox_input(id, label, type = "toggle", is_marked = is_marked, style = style)
```
[the remark this answers is not in the document]
Comment (dokato) [reply R/checkbox.R:52]: hmm, here I don't understand what's wrong, for me works as charm, also we don't really use `shinyjs` here

File: R/date_input.R [hunk 14]
```diff
@@ -1,6 +1,6 @@
 #' Define simple date input with semantic ui styling
```
[the remark this answers is not in the document]
Comment (dokato) [reply R/date_input.R:1]: it should be changed to fomantic at some point anyway probalby

File: R/dsl.R (hunk 3 above)
Comment (dokato) [reply R/dsl.R:428]: here again not sure what you're referring too, this example worked for me

File: R/utils.R (hunk 11 above)
Comment (dokato) [reply R/utils.R:27]: This is hidden function, not to see by the user, actually examples of hidden function cause problem when doing checks for CRAN, but I elaborated in the description.

File: R/utils.R (hunk 12 above)
Comment (dokato) [reply R/utils.R:39]: same as above

**Comment by dokato** [comment]
@olgamie I tried to also fix examples from `examples/` folder. Hope I covered all.

File: R/dsl.R [hunk 15]
```diff
@@ -17,40 +17,40 @@
 #'   shinyUI(
 #'     semanticPage(
 #'       # Basic icon
-#'       uiicon("home"),
+#'       icon("home"),
```
[the remark this answers is not in the document]
Comment (olgamie) [reply R/dsl.R:20]: Unfortunately there is still a problem with rendering icons properly:
image

File: R/checkbox.R (hunk 13 above)
[the remark this answers is not in the document]
Comment (olgamie) [reply R/checkbox.R:52]: So I had to add shiny.semantic::toggle to get the example work. It seems that previous error was due to toggle function coming from shinyjs.

```
library(shiny)
library(shiny.semantic)

ui <- semanticPage(
   shinyUI(
      shiny.semantic::toggle("tog1", "My Label", TRUE)
   )
)
server <- function(input, output, session) {
   observeEvent(input$tog1, {
      print(input$tog1)
   })
}
shinyApp(ui, server)
```

**Comment by dokato** [comment]
Thanks @olgamie for catching these bugs. Icons example seem to be working for me, look below. `app.R` indeed was broken, fixed now. Export added too.

I introduced also some fixes to dropdown and update action button and there were some minor mistakes in roxygen compilation.
 
Screenshot 2020-07-21 at 12 28 50

**Comment by dokato** [comment]
@olgamie done

**Pull request merged**